org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/general/completion/FilePathCompletionParticipant.java

```diff
@@ -259,4 +262,13 @@ private static void createFilePathCompletionItem(File f, Range replaceRange, ICo
 		response.addCompletionItem(item);
 	}
 
+	private static boolean hasPathBeginning(String currentText) {
+		return currentText.startsWith("/")
+		|| currentText.matches("^[A-Za-z]:\\\\.*")

```

Comment:
Is it possible to do that without RegeExp?

Reply:
At the very least I change this to a `Pattern`. I think that a RegExp is the cleanest solution, but I could also try to do something with looping over the characters.

Reply:
Indeed the first improvement is to cache the regexp in Pattern. My fear is this code will be called a lot when user will type some content in attribute and we shoudl have good performance and not some memory problem.

Reply:
I switched to using a function instead of a regex